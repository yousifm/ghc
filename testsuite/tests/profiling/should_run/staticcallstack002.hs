{-# LANGUAGE UnboxedTuples #-}
module Main where

import GHC.Stack.CCS

-- Unboxed data constructors don't have info tables so there is
-- a special case to not generate distinct info tables for unboxed
-- constructors.
main = do
  print . tail =<< whereFrom (undefined (# #))
  print . tail =<< whereFrom (undefined (# () #))
  print . tail =<< whereFrom (undefined (# (), () #))
  print . tail =<< whereFrom (undefined (# | () #))

