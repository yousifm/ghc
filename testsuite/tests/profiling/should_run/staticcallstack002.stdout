["15","Any","main","Main","staticcallstack002.hs:10:30-46"]
["15","Any","main","Main","staticcallstack002.hs:11:30-49"]
["15","Any","main","Main","staticcallstack002.hs:12:30-53"]
["15","Any","main","Main","staticcallstack002.hs:13:30-51"]
